packages/react/roving-focus/src/RovingFocusGroup.tsx

```diff
@@ -1,144 +1,237 @@
+      onInteract={React.useCallback(() => setHasInteracted(true), [])}
+      onItemFocus={React.useCallback((tabStopId) => setCurrentTabStopId(tabStopId), [
+        setCurrentTabStopId,
+      ])}
+      onItemShiftTab={React.useCallback(() => setIsTabbingBackOut(true), [])}
     >
-      {children}
+      <Primitive
+        tabIndex={isTabbingBackOut ? -1 : 0}
+        aria-orientation={orientation}
+        data-orientation={orientation}
+        {...groupProps}
+        as={as}
+        ref={composedRefs}
+        onFocus={composeEventHandlers(props.onFocus, (event) => {
+          if (!isTabbingBackOut && event.target === event.currentTarget) {
+            const entryFocusEvent = new Event(ENTRY_FOCUS, EVENT_OPTIONS);
+            event.currentTarget.dispatchEvent(entryFocusEvent);
+
+            if (!entryFocusEvent.defaultPrevented) {
+              const items = Array.from(itemMap.values()).filter((item) => item.focusable);
+              const activeItem = items.find((item) => item.active);
+              const currentItem = hasInteracted
+                ? items.find((item) => item.id === currentTabStopId)
+                : undefined;
+              const candidateItems = [activeItem, currentItem, ...items].filter(Boolean) as Item[];
+              const candidateNodes = candidateItems.map((item) => item.ref.current!);
+              focusFirst(candidateNodes);
+            }
+          }
+        })}
+        onBlur={composeEventHandlers(props.onBlur, () => setIsTabbingBackOut(false))}
+      />
     </RovingFocusProvider>
   );
-}
+}) as RovingFocusGroupPrimitive;
 
 RovingFocusGroup.displayName = GROUP_NAME;
 
 /* -------------------------------------------------------------------------------------------------
- * useRovingFocus
+ * RovingFocusItem
  * -----------------------------------------------------------------------------------------------*/
+
 const ITEM_NAME = 'RovingFocusItem';
+const ITEM_DEFAULT_TAG = 'span';
 
-type UseRovingFocusItemOptions = { disabled?: boolean; active?: boolean };
+type RovingFocusItemOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    focusable?: boolean;
+    active?: boolean;
+  }
+>;
 
-function useRovingFocus({ disabled, active = false }: UseRovingFocusItemOptions) {
-  const itemId = useId();
+type RovingFocusItemPrimitive = Polymorphic.ForwardRefComponent<
+  typeof ITEM_DEFAULT_TAG,
+  RovingFocusItemOwnProps
+>;
+
+const RovingFocusItem = React.forwardRef((props, forwardedRef) => {
+  const { as = ITEM_DEFAULT_TAG, focusable = true, active = false, ...itemProps } = props;
+  const id = useId();
+  const ref = React.useRef<HTMLElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
   const context = useRovingFocusContext(ITEM_NAME);
-  const isTabStop = itemId === context.tabStopId;
+  const isCurrentTabStop = context.currentTabStopId === id;
+  const isCurrentTabStopRef = React.useRef(isCurrentTabStop);
```

Comment:
Do we still need this ref? I can't see it being used.

Reply:
Great catch, no needed anymore, forgot to delete when we removed all the other code.